data/pilots/rebel-alliance/t-65-x-wing.json

```diff
@@ -307,6 +307,262 @@
       "extended": true,
       "keywords": ["X-wing"],
       "epic": true
+    },
+    {
+      "name": "Biggs Darklighter",
+      "caption": "Battle of Yavin",
+      "initiative": 3,
+      "limited": 1,
+      "cost": 5,
+      "standardLoadout": true,
+      "xws": "biggsdarklighter-battleofyavin",
+      "ability": "During the System Phase, you may choose 1 friendly ship at range 1. If you do, treat your initiative as equal to the chosen ship's initiative until the end of the Activation Phase.",
+      "shipAbility": {
+        "name": "Hope",
+        "text": "After another friendly ship at range 0-3 is destroyed, you may perform a [Focus] or [Boost] action."
+      },
+      "upgrades": [
+        {
+          "title": "Attack Speed",
+          "slots": ["Talent"],
+          "ability": "After you fully execute a [3 [Straight]] or [4 [Straight]] maneuver, you may perform a boost using the [1 [Straight]] template. (This is not an action)."
+        },
+        {
+          "title": "Selfless",
+          "slots": ["Talent"],
+          "ability": "While another friendly ship at range 0-1 defends, before the Neutralize Results step, if you are in the attack arc, you may suffer 1 [Critical Hit] damage to cancel 1 [Critical Hit] result.",
+          "limited": 1
+        },
+        {
+          "title": "Proton Torpedoes",
+          "slots": ["Torpedo"],
+          "ability": "Attack ([Lock]): Spend 1 [Charge]. Change 1 [Hit] result to a [Critical Hit] result.",
+          "charges": { "value": 2, "recovers": 0 },
+          "attack": {
+            "arc": "Front Arc",
+            "value": 4,
+            "minrange": 2,
+            "maxrange": 3,
+            "ordnance": true
+          }
+        },
+        {
+          "title": "R2-F2",
+          "slots": ["Astromech"],
+          "ability": "After you reveal your dial, you may spend 1 [Charge] and gain 1 disarm token to recover 1 shield.",
+          "charges": { "value": 2, "recovers": 0 },
+          "limited": 1
+        }
+      ],
+      "slots": [
```

Comment:
Having an empty set of slots feels weird, but again thought it was probably better to include for compatibility.

Reply:
This makes sense. Ironically, we'll still have issues with ships with empty slots (generic empire TIE Fighters all have 0 loadout and no upgrade bar), but I think it's clear what's going on at that point then.

Reply:
Yep, empty slots is valid and means "this pilot has no upgrade slots". That's different to `standardLoadout` pilots which ideally should have no slots field, because the concept of slots isn't valid on that kind of pilot.

If the conditional stuff in the schema is too difficult then we can just accept an empty slots entry for standardLoadout pilots, but the conditional stuff would be really helpful to future contributors.

Reply:
I updated the schema to accept upgrades, but haven't done the oneOf for upgrades/slots. I'll get it in the first round of typo fixes.